### PR TITLE
Add EKS Prometheus Integration Test

### PR DESCRIPTION
**Pull request by JayPolanco**
# Description of the issue
Adds a new github action for testing prometheus on EKS

# Description of changes
- Adds an eks_deployment_matrix to generate matrix for the eks prometheus test.
- Creates a EKSPrometheus github action for testing prometheus on eks. 

# License
By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

# Tests
Tested in a branch of the main repo

# Requirements
_Before commit the code, please do the following steps._
1. Run `make fmt` and `make fmt-sh`
2. Run `make linter`

### CODE REVIEW

File: .github/workflows/integration-test.yml [hunk 1]
```diff
@@ -1010,6 +1013,79 @@ jobs:
...
+            terraform init
+            if terraform apply --auto-approve \
+              -var="test_dir=${{ matrix.arrays.test_dir }}"\
+              -var="cwagent_image_repo=${{ steps.login-ecr.outputs.registry }}/${{ env.ECR_INTEGRATION_TEST_REPO }}" \
+              -var="cwagent_image_tag=${{ github.sha }}" \
+              -var="k8s_version=${{ matrix.arrays.k8s_version }}"; then
+              terraform destroy -auto-approve
+            else
+              terraform destroy -auto-approve && exit 1
+            fi
```
Comment (SaxyPandaBear) [inline .github/workflows/integration-test.yml:1078]: So with the changes from the ECS/EKS statsd templating, would it make more sense for us to add to that test generation, rather than duplicate the GitHub Action? @movence

Comment (JayPolanco) [reply .github/workflows/integration-test.yml:1078]: That was something I considered, but decided not to in order to make a clear distinction between the deployment types as those use daemon and this one uses deployment. We could change it to be one large integration test, but I'd image testing time would increase and I don't know if that trade off is worth it

Comment (movence) [reply .github/workflows/integration-test.yml:1078]: I think one EKS integration test `EKSIntegrationTest` makes sense here since now we can specify `terraformDir` to have tests with different infra configs. Envisioning one EKS github action with all tests under it as jobs including statsd, prometheus and etc. It feels like we don't fully utilize the benefits of having `terraformDir` and `TestSuite` when each test types get separated out to their own.

**Review by SaxyPandaBear — APPROVED** [review]

**Comment by movence** [comment]
LGTM we can revisit the structure later if needed

**Review by movence — APPROVED** [review]
